micro-rdk/src/common/ota.rs

```diff
@@ -159,11 +158,30 @@ impl OtaService {
                 )))
             }
         };
+
+        #[cfg(not(feature = "esp32"))]
+        let max_size = 1024 * 1024 * 4; // 4MB
+        #[cfg(feature = "esp32")]
+        let max_size = {
+            log::debug!("getting size of dst partition...");
+            let ptr: *const esp_partition_t =
+                unsafe { esp_ota_get_next_update_partition(std::ptr::null()) };
+            if ptr.is_null() {
+                return Err(OtaError::Other(
+                    "failed to get dst partition info - may not exist".to_string(),
```

Comment:
it will definitely have its own `OtaError` type. that's the final major ticket is doing a full sweep and transpose of all errors into specific `OtaErrors`.

If this fails then the `OtaService::from_config` would return an error. But I can see if there's something specific I can check at the start to confirm that `otadata` exists